plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -250,6 +251,29 @@ public void setColumnComment(SchemaTableName schemaTableName, String columnName,
         tableCache.invalidate(schemaTableName);
     }
 
+    public void renameTable(SchemaTableName oldName, SchemaTableName newName)
+    {
+        String oldSchemaName = toRemoteSchemaName(oldName.getSchemaName());
+        String oldTableName = toRemoteTableName(oldSchemaName, oldName.getTableName());
+        String newSchemaName = toRemoteSchemaName(newName.getSchemaName());
+
+        // Need to check explicitly because the old collection may not exist when it doesn't have any data
+        if (collectionExists(client.getDatabase(oldSchemaName), oldTableName)) {
+            getCollection(oldName).renameCollection(new MongoNamespace(newSchemaName, newName.getTableName()));
+        }
+
+        // Schema collection should always have the source table definition
+        MongoCollection<Document> oldSchema = client.getDatabase(oldSchemaName).getCollection(schemaCollection);
+        Document tableDefinition = oldSchema.findOneAndDelete(new Document(TABLE_NAME_KEY, oldTableName));
+        requireNonNull(tableDefinition, "Table definition not found in schema collection: " + oldTableName);
```

Comment:
at some point we should consider introducing a mechanism to apply compensation actions in case of partial failure of methods like these so that the operations are "transactional".